File: 1_contest/1_task.cpp
#include <iostream>
#include <vector>


using namespace std;

bool binSearch(vector<int> &arr, int x) {
    int l = 0, r = arr.size() - 1;
    
    while (l <= r) {
        int m = l + (r - l) / 2;
        
        if (arr[m] == x) {
            return true;
        } 
        
        if (arr[m] < x) {
            l = m + 1;
        } else {
            r = m - 1;
        }
    }
    return false;
}

int main() {
    int n, m;
    cin >> n >> m;
    vector<int> arr(n);

    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for (int i = 0; i < m; i++) {
        int x;
        cin >> x;
        if (binSearch(arr, x)) {
            cout << "YES ";
        } else {
            cout << "NO ";
        }
    }

    cout << '\n';
    return 0;
}

File: 1_contest/1_task.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

func binSearch(arr []int, x int) bool {
	l, r := 0, len(arr)-1
	for l <= r {
		m := l + (r-l)/2
		if arr[m] == x {
			return true
		}
		if arr[m] < x {
			l = m + 1
		} else {
			r = m - 1
		}
	}
	return false
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	next := func() int {
		scanner.Scan()
		val, _ := strconv.Atoi(scanner.Text())
		return val
	}

	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	n := next()
	m := next()

	arr := make([]int, n)
	for i := 0; i < n; i++ {
		arr[i] = next()
	}

	for i := 0; i < m; i++ {
		x := next()
		if binSearch(arr, x) {
			writer.WriteString("YES ")
		} else {
			writer.WriteString("NO ")
		}
	}
	writer.WriteString("\n")
}
